### PR TITLE
Fix for "Jump to" button overlapping Logout button on the Profile screen

### LINKED ISSUE #18893: Jump to button overlaps Logout button on the Profile screen
# Bug Report

## Problem 
[comment]: # (A problem description)
User still can tap a Logout but need to find a free space on the Logout button for this purpose

#### Expected behavior
Buttons look as designed [here](https://www.figma.com/file/PiN0uwRgZj4L9U4N7xO7Xb/Profile-for-Mobile?type=design&node-id=3773-221806&mode=design&t=6VjQ1LSVnDiMyB2y-0)
Screenshot 2024-02-19 at 09 57 11

#### Actual behavior
Screenshot 2024-02-19 at 09 58 14

### Reproduction

1. Login to Status app
2. Go to Profile
3. Scroll down to Logout button

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS
- Device: Samsung A34, iPhone SE

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by alwx**
Fixes #18893

#### Platforms

- Android
- iOS

##### Functional

- settings screen

status: ready

### CODE REVIEW

**Comment by J-Son89** [comment]
Hey @alwx, if possible a video or screen shots would be very helpful here 👍

File: src/status_im/contexts/profile/settings/view.cljs [hunk 1]
```diff
@@ -78,7 +77,7 @@
         :customization-color customization-color
         :label               (i18n/label :t/jump-to)}}
       {:position :absolute
-       :bottom   jump-to.constants/floating-shell-button-height}]]))
+       :bottom   20}]]))
```
Comment (Parveshdhull) [inline src/status_im/contexts/profile/settings/view.cljs]: Thank you @alwx for fixing this issue.

Please don't hardcode 20, I think we can use `(:bottom insets) here. wdyt?

Comment (Parveshdhull) [reply src/status_im/contexts/profile/settings/view.cljs]: Also, if possible Please also make same change in `status-im.contexts.profile.contact.view`. (It seems they are copy pasted, and using bottom wrong)

File: src/status_im/contexts/profile/settings/style.cljs [hunk 2]
```diff
@@ -6,8 +6,8 @@
...
-(def footer-container
-  {:margin-top         8
-   :margin-bottom      50
-   :padding-horizontal 20
-   :padding-vertical   12})
+(defn footer-container
+  [bottom]
+  {:padding-horizontal 20
+   :padding-top        20
+   :padding-bottom     (- 78 bottom)})
```
Comment (Parveshdhull) [inline src/status_im/contexts/profile/settings/style.cljs]: - Its 77 in design.
- I think we should not subtract this bottom. 77 is used to make sure log out don't overlap jump-to button.
 So, jump-to button top position will be (21(bottom padding) + 44 (jump-to button)  = 65

(So if we remove this bottom, bottom padding become = 77-21 = 56)

which is lower than, jump-to position.

**Review by Parveshdhull — CHANGES_REQUESTED** [review]

File: src/status_im/contexts/profile/settings/view.cljs (hunk 1 above)
Comment (alwx) [reply src/status_im/contexts/profile/settings/view.cljs]: `(:bottom insets)` has a different value, on Androids it can be zero, on iPhones with Face ID it's 34. 
I would say it needs to be *at least* 20, and on iPhones it should be even higher just because the bottom part of the screen is already occupied with a pill that's used for multitasking / minimizing the app.

File: src/status_im/contexts/profile/settings/style.cljs (hunk 2 above)
Comment (alwx) [reply src/status_im/contexts/profile/settings/style.cljs]: Here the idea is that it's at least 78 (or 77, whatever). If the bottom inset is set, like on iPhones, then it will be lower which is fine because the bottom inset is not part of the calculation anyway.

File: src/status_im/contexts/profile/settings/view.cljs (hunk 1 above)
Comment (Parveshdhull) [reply src/status_im/contexts/profile/settings/view.cljs]: > I would say it needs to be at least 20, and on iPhones it should be even higher just because the bottom part of the screen is already occupied with a pill that's used for multitasking / minimizing the app.

Yes, that's what bottom insets will provide. Height of that multitasking area.

image

> on Androids it can be zero, 

what are issues with that? Even if is zero, it is position of floating-shell-button container as same as figma.
And jump-to position will not be zero(because container already position jump-to close to the top).

If we keep this bottom padding also for android by hard coding where we don't have multitasking button then we are unnecessarily increasing this space. Similar to  screenshot shared in issue.

image

Comment (alwx) [reply src/status_im/contexts/profile/settings/view.cljs:80]: I don't get what you're talking about — the bottom inset on latest iPhones is 34 and not 21.

> Even if is zero, it is position of floating-shell-button container as same as figma.

I cannot really parse this sentence, sorry. Could you rephrase it to make it clear what you mean?

Comment (alwx) [reply src/status_im/contexts/profile/settings/view.cljs:80]: Or actually don't bother much — I will push changes in the next hour or so anyway.

**Comment by alwx** [comment]
@Parveshdhull just updated it all — now it completely matches the design and probably also follows the logic that you were describing above. The only important difference is that the bottom inset on iOS is actually a bit bigger than the height of the visible "pill" itself — in reality it's not 21 like it's specified in our designs but 34 which requires us to apply an extra offset on iOS.
Tested it all on both simulator and real device, both Android and iOS.

File: src/status_im/contexts/profile/settings/style.cljs (hunk 2 above)
Comment (alwx) [reply src/status_im/contexts/profile/settings/style.cljs]: @Parveshdhull https://github.com/status-im/status-mobile/pull/19318#issuecomment-2009695657

File: src/status_im/contexts/profile/settings/style.cljs [hunk 3]
```diff
@@ -1,13 +1,27 @@
...
-   :padding-vertical   12})
+(def ^:const jump-to-button-height 44)
+(def ^:const footer-padding 20)
+(def ^:const ios-bottom-offset -10)
+
+(defn footer-container
+  [bottom]
+  {:padding-horizontal footer-padding
+   :padding-top        footer-padding
+   :padding-bottom     (+ jump-to-button-height
```
Comment (Parveshdhull) [inline src/status_im/contexts/profile/settings/style.cljs]: Please directly use `jump-to.constants/floating-shell-button-height`, so that we needs to change only at single place, if needed.

**Review by Parveshdhull — APPROVED** [review]
Thank you very much for fixing the issue and verifying.

**Review by mohsen-ghafouri — APPROVED** [review]

**Review by Francesca-G — APPROVED** [review]
Looks good to me ✨
